app/controllers/students/base_controller.rb

```diff
@@ -8,7 +8,7 @@ def must_be_student
   end
 
   def current_team
-    @current_team ||= current_user.teams.first
+    @current_team ||= current_user.teams.last
```

Comment:
While you can only access the controller as a student, it might be possible that you have once been a student but your most recent team membership is as a supervisor or coach. That way around `current_user.(any_team).in_current_season` could yield an irrelevant team, ie. one where you are not a student on that team.